src/core/server.cc

```diff
@@ -109,6 +113,90 @@ InferenceServer::InferenceServer()
   inflight_request_counter_ = 0;
 }
 
+void
+InferenceServer::PrintSummary()
+{
+  // Models Summary
+  auto model_states = model_repository_manager_->BackendStates();
+
+  std::vector<std::string> model_headers;
+  model_headers.emplace_back("Model");
+  model_headers.emplace_back("Version");
+  model_headers.emplace_back("Status");
+
+  std::unique_ptr<triton::common::TablePrinter> models_table;
+  triton::common::TablePrinter::Create(&models_table, model_headers);
+
+  for (const auto& model_state : model_states) {
+    std::vector<std::string> model_record;
+    auto model_version_map = model_state.second;
+    std::string model_name = model_state.first;
+
+    // If model_version_map size is zero, no version is found for this model
+    if (model_version_map.size() == 0) {
+      model_record.emplace_back(model_name);
+      model_record.emplace_back("-");
+      model_record.emplace_back("Not loaded: No model version was found");
+      models_table->insert_row(model_record);
+    } else {
+      for (const auto& model_map : model_version_map) {
+        std::string model_version = std::to_string(model_map.first);
+        auto model_status_pair = model_map.second;
+        std::string model_status =
+            ModelReadyStateString(model_status_pair.first);
+
+        if (model_status_pair.second != "") {
+          model_status += ": " + model_status_pair.second;
+        }
+
+        model_record.emplace_back(model_name);
+        model_record.emplace_back(model_version);
+        model_record.emplace_back(model_status);
+        models_table->insert_row(model_record);
+      }
+    }
+  }
+  std::unique_ptr<std::string> models_table_string =
+      models_table->print_table();
+  LOG_INFO << *models_table_string;
+
+  // Backends Summary
+  std::unique_ptr<triton::common::TablePrinter> backends_table;
+
+  std::vector<std::string> backend_headers;
+  backend_headers.emplace_back("Backend");
+  backend_headers.emplace_back("Path");
+  backend_headers.emplace_back("Config");
+
+  triton::common::TablePrinter::Create(&backends_table, backend_headers);
+  TritonBackendManager& triton_backend_manager =
+      TritonBackendManager::Instance();
+
+  std::lock_guard<std::mutex> lock(triton_backend_manager.Mutex());
+  auto backend_map = triton_backend_manager.BackendMap();
+
+  for (const auto& backend_pair : backend_map) {
+    std::vector<std::string> backend_record;
+    auto backend = backend_pair.second.lock();
+    if (backend != nullptr) {
+      backend_record.emplace_back(backend->Name());
+      const char* config_message;
+      size_t config_message_size;
+      backend->BackendConfig().Serialize(&config_message, &config_message_size);
```

Comment:
Do we want to print out the entire config message? I think this is necessary only if the model config is autofilled.

Reply:
Usually, backends do not have a lot of config flags. I can add an if statement to print it only if it is being auto-filled. 

/cc @deadeyegoodwin

Reply:
I think @CoderHam  was confusing the backend config with model config. I think what you have it ok.

Reply:
That's right I was mistaken. Marking this as resolved.